banzai_nres/classify.py

```diff
@@ -55,23 +56,87 @@ def find_object_in_catalog(image, db_address, gaia_class, simbad_class):
     else:
         simbad = import_utils.import_attribute(simbad_class)
         simbad_connection = simbad()
-        simbad_connection.add_votable_fields('pmra', 'pmdec', 'fe_h')
+        simbad_connection.add_votable_fields('pmra', 'pmdec', 'fe_h', 'otype')
         try:
             results = simbad_connection.query_region(coordinate, radius='0d0m10s')
+            results = restrict_simbad_results_to_stellar_only(results)
+            results = parse_simbad_coordinates(results)
+            results = convert_simbad_coordinates_to_degrees(results)
+            # note that results will be None if coordinate conversion fails
         except astroquery.exceptions.TableParseError:
             response = simbad_connection.last_response.content
             logger.error(f'Error querying SIMBAD. Response from SIMBAD: {response}', image=image)
             results = None
+
         if results:
-            image.classification = dbs.get_closest_phoenix_models(db_address, results[0]['Fe_H_Teff'],
-                                                                  results[0]['Fe_H_log_g'])[0]
+            results = get_closest_source(results, coordinate)
+            image.classification = dbs.get_closest_phoenix_models(db_address, results['Fe_H_Teff'],
+                                                                  results['Fe_H_log_g'])[0]
+            image.pm_ra = results['PMRA']  # note that we always assume these are in mas/yr... which they should be.
+            image.pm_dec = results['PMDEC']
             # Update the ra and dec to the catalog coordinates as those are basically always better than a user enters
             # manually.
-            image.ra, image.dec = results[0]['RA'], results[0]['DEC']
-            image.pm_ra, image.pm_dec = results[0]['PMRA'], results[0]['PMDEC']
+            image.ra = results['RA']
+            image.dec = results['DEC']
         # If there are still no results, then do nothing
 
 
+def get_closest_source(results, coordinate):
+    ra_unit, dec_unit = results['RA'].unit, results['DEC'].unit
+    coords = [SkyCoord(row['RA'], row['DEC'], unit=(ra_unit, dec_unit)) for row in results]
+    diffs = [coord.separation(coordinate).to(units.arcsecond).value for coord in coords]
+    closest = np.argmin(np.abs(diffs))
+    return results[closest]
+
+
+def restrict_simbad_results_to_stellar_only(results):
+    # remove galaxies
+    results = results[['*' in i for i in results['MAIN_ID']]]
+    is_nonstellar = np.zeros(len(results), dtype=bool)
+    nonstellar_otypes = ['su*', 'Pl?', 'Pl', 'SN', 'Galaxy', 'Planet', 'Planet?']
```

Comment:
Doesn't line 94 already sort out nonstellar types, since it just selects otypes containing the character *? Why is it necessary to do this additional search for otypes other than su*?

Reply:
Good question. It is because planets still show up with MAINID beginning with *. E.g. for tau ceti:
```
* tau Cet b  ...  Planet?
* tau Cet h   ...  Planet
```